File: ajps-backend/src/main/java/com/himusharier/ajps_backend/model/Journal.java
package com.himusharier.ajps_backend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "ajps_journals")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Journal {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String journalName;

    @Column(nullable = false, unique = true)
    private String issn;

    @Column(nullable = false)
    private String frequency;

    @Column(nullable = false)
    private String journalType;

    @Column(nullable = false, unique = true)
    private String journalCode;

    @Column(nullable = false)
    private String contactEmail;

    @Column(nullable = false)
    private String journalUrl;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String aimsScopes;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String aboutJournal;

    @Column(name = "cover_image_url")
    private String coverImageUrl;

    @OneToMany(mappedBy = "journal", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    private List<EditorAssignment> editorAssignments = new ArrayList<>();

    // NEW: One-to-Many relationship with Issue
    @OneToMany(mappedBy = "journal", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference("journal-issue")
    private List<Issue> issues = new ArrayList<>();
}

File: ajps_flutter_app/lib/models/journal.dart
import 'package:ajps_flutter_app/models/issue.dart';

class Journal {
  final int? id;
  final String journalName;
  final String journalUrl;
  final String issn;
  final String frequency;
  final String journalType;
  final String coverImageUrl;
  final String aboutJournal;
  final String aimsScopes;
  final Issue? issues;

  Journal({
    this.id,
    required this.journalName,
    required this.journalUrl,
    required this.issn,
    required this.frequency,
    required this.journalType,
    required this.coverImageUrl,
    required this.aboutJournal,
    required this.aimsScopes,
    this.issues,
  });

  factory Journal.fromJson(Map<String, dynamic> json) {
    return Journal(
      id: json['id'] is String ? int.tryParse(json['id']) : json['id'],
      journalName: json['journalName'],
      journalUrl: json['journalUrl'],
      issn: json['issn'],
      frequency: json['frequency'],
      journalType: json['journalType'],
      coverImageUrl: json['coverImageUrl'],
      aboutJournal: json['aboutJournal'],
      aimsScopes: json['aimsScopes'],
      issues: json['issues'],
    );
  }
}
